File: PersonTbc.Data/Models/Person.cs
using System.ComponentModel.DataAnnotations;

namespace PersonTbc.Data.Models;

public class Person : BaseModel<int>
{
    [Required] public string FirstName { get; set; }
    [Required] public string LastName { get; set; }
    [Required] public ulong PersonalId { get; set; }
    public DateTime? DateOfBirth { get; set; }

    [Required] public int GenderId { get; set; }
    public GenderEntity Gender { get; set; }
    [Required] public AccountStatus Status { get; set; } = AccountStatus.Active;

    public DateTime CreatedDate { get; set; } = DateTime.Now;
}

File: angular-client/src/app/shared/interfaces/Person.ts
export interface Person {
  id?: number
  firstName: string;
  lastName: string;
  personalId: number;
  dateOfBirth?: string;
  gender: string;
  accountStatus: string;
}

export interface UpdatePerson {
  id: number
  firstName: string;
  lastName: string;
  personalId: number;
  dateOfBirth?: string;
  gender: string;
  status: number;
}

export interface Response {
  statusCode: number,
  error?: string,
  data?: object
}

export interface AuthResponse {
  statusCode: number,
  error?: string,
  token?: string
}

